streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -272,25 +329,35 @@ public void print(final Printed<K, V> printed) {
         to(topic, producedInternal);
         return builder.stream(Collections.singleton(topic),
                               new ConsumedInternal<>(producedInternal.keySerde(),
-                                            producedInternal.valueSerde(),
-                                            new FailOnInvalidTimestamp(),
-                                            null));
+                                                     producedInternal.valueSerde(),
+                                                     new FailOnInvalidTimestamp(),
+                                                     null));
     }
 
     @Override
     public void foreach(final ForeachAction<? super K, ? super V> action) {
         Objects.requireNonNull(action, "action can't be null");
         String name = builder.newProcessorName(FOREACH_NAME);
 
-        builder.internalTopologyBuilder.addProcessor(name, new KStreamPeek<>(action, false), this.name);
+        ProcessorParameters processorParameters = new ProcessorParameters<>(new KStreamPeek<>(action, false), name);
+
+        StatelessProcessorNode<K, V> foreachNode = new StatelessProcessorNode<>(name,
+                                                                                processorParameters,
+                                                                                false);
```

Comment:
Why we can pass `false` for `repartitionRequired` here??

Reply:
ack, fixed